confgenerator/testdata/valid/linux/logging-processor_exclude_logs/golden_fluent_bit_main.conf

```diff
@@ -0,0 +1,774 @@
+@SET buffers_dir=/var/lib/google-cloud-ops-agent/fluent-bit/buffers
+@SET logs_dir=/var/log/google-cloud-ops-agent/subagents
+
+[SERVICE]
+    Daemon                    off
+    Flush                     1
+    HTTP_Listen               0.0.0.0
+    HTTP_PORT                 2020
+    HTTP_Server               On
+    Log_Level                 info
+    storage.backlog.mem_limit 50M
+    storage.checksum          on
+    storage.max_chunks_up     128
+    storage.metrics           on
+    storage.sync              normal
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/default_pipeline_syslog
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              /var/log/messages,/var/log/syslog
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               default_pipeline.syslog
+    storage.type      filesystem
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/p1_sample_logs
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              /tmp/*.log
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               p1.sample_logs
+    storage.type      filesystem
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/ops-agent-fluent-bit
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              ${logs_dir}/logging-module.log
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               ops-agent-fluent-bit
+    storage.type      filesystem
+
+[FILTER]
+    Add   logging.googleapis.com/logName syslog
+    Match default_pipeline.syslog
+    Name  modify
+
+[FILTER]
+    Match      p1.sample_logs
+    Name       nest
+    Nest_under record
+    Operation  nest
+    Wildcard   *
+
+[FILTER]
+    Condition Key_value_matches $record['message'] (?i)^bar$
+    Match     p1.sample_logs
+    Name      modify
+    Set       __match_p1_sample_logs 1
+
+[FILTER]
+    Exclude __match_p1_sample_logs 1
+    Match   p1.sample_logs
+    Name    grep
+
+[FILTER]
+    Match           p1.sample_logs
+    Name            modify
+    Remove_wildcard __match_p1_sample_logs
+
+[FILTER]
+    Match        p1.sample_logs
+    Name         nest
+    Nested_under record
+    Operation    lift
+
+[FILTER]
```

Comment:
Starring this line here, the next many lines are for another exclusion filter for p1.sample_logs. Have we considered optimization, e.g. when the exclusion filters rules target on the same tag e.g. `p1.sample_logs`, we can aggregate it into one Regex instead of repeating the steps as that also requires more resources.

Reply:
This will mainly be driven by performance testing; if it reveals any compelling trade-offs then we'd consider doing optimizations. That testing will be done soon & finished before we merge anything here.

Reply:
Looks like LUA filter based solution might have a slightly better performance. Add we add a `potential future improvement` (TODO in code and/or file an internal tracking ticket for it?)

Reply:
Added a comment to the code and filed an internal tracking ticket.